torchmetrics/metric.py

```diff
@@ -447,13 +486,28 @@ def _load_from_state_dict(
     ) -> None:
         """ Loads metric states from state_dict """
 
-        # only global rank 0 should be reloading the values present in the ``state_dict``
-        # as the state contains synced values across all progress_group
+        # assumption: users didn't manually sync the Metric states by default.
+        has_synced = state_dict.pop(prefix + "has_synced", None)
```

Comment:
just to double check, is it ok to pop? 
the intention is not to re-use the state dict somewhere else?
could the user want to load the same state dict onto different metrics, like they could do in pytorch with model weights?

Reply:
Yes, you need to pop or it will raise an error.

Reply:
That's the expected behaviour from PyTorch. Checkpoint are being consumed.